pkg/autoscaler/autoscaler.go

```diff
@@ -181,7 +181,7 @@ func (agg *perPodAggregation) usageRatio(now time.Time) float64 {
 		return float64(1.0)
 	}
 	outOfService := now.Sub(*agg.lameduckTime)
-	return float64(1.0) - (float64(outOfService) / float64(agg.window))
+	return math.Max(0.0, float64(1.0) - (float64(outOfService) / float64(agg.window)))
```

Comment:
Can this even go below 1.0? `outOfService` should never be larger than the `agg.window` and thus the division of both should never be larger than 1?

Reply:
> `outOfService` should never be larger than the `agg.window`
Why? I think this would happen in panic mode. Cause we only delete the stale stats after 60s. And it is possible that the queue-proxy takes more than the PanicWindow time to stop sending stats.

Reply:
That's only my guess. If we keep sending requests to the service which some pod is start to terminate. Then it is possible that dispatch the request to the terminating pod, and then this request would have the chance that can't get the right response.

Reply:
FWIW: We (as in knative) does not control which Pods gets the request in the end. We're not even killing specific pods. We rather tell a deployment to scale down, and then the deployment decides which pods to kill.

Likewise with requests to pods. Istio's proxies will decide where to send the requests, we have no real influence on that.

Reply:
After fixing the problem of #2404, I think my assumption will not happened. Since the ratio will never blow 0, then i think it is np to left as it is.